source/features/hide-useless-comments.tsx

```diff
@@ -55,11 +57,112 @@ function unhide(event: React.MouseEvent<HTMLButtonElement>): void {
 	event.currentTarget.parentElement!.remove();
 }
 
+async function getMutedUsers(): Promise<string[]> {
+	return ((await optionsStorage.getAll() as Options).mutedUsers).split(' ');
+}
+
+async function setMutedUsers(mutedUsers: string[]): Promise<void> {
+	return optionsStorage.set({mutedUsers: mutedUsers.join(' ')});
+}
+
+function minimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.remove('d-none');
+	select('.minimized-comment .Details-element', comment)!.removeAttribute('open');
+	select('.unminimized-comment', comment)!.classList.add('d-none');
+}
+
+function unminimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.add('d-none');
+	select('.minimized-comment .Details-element', comment)!.removeAttribute('open');
+	select('.unminimized-comment', comment)!.classList.remove('d-none');
+}
+
+async function onMuteUnmuteClick(event: React.MouseEvent<HTMLButtonElement>): Promise<void> {
+	let mutedUsers = await getMutedUsers();
+
+	const comment = (event.target as HTMLElement).closest('.js-comment-container')!;
+	const viewportOffset = comment.getBoundingClientRect().top;
+	const user = select('.author', comment)!.textContent!;
+
+	if (mutedUsers.includes(user)) {
+		mutedUsers = mutedUsers.filter(mutedUser => mutedUser !== user);
+	} else {
+		mutedUsers.push(user);
+	}
+
+	await setMutedUsers(mutedUsers);
+
+	const avatars = select.all(`.js-discussion .js-comment-container .avatar[alt="@${user}"`);
+	const comments = avatars.map(avatar => avatar.closest('.js-comment-container') as HTMLElement);
+
+	for (const comment of comments) {
+		if (mutedUsers.includes(user)) {
+			minimizeComment(comment);
+			for (const button of select.all('.rgh-mute-unmute-button', comment)) {
+				button.textContent = 'Unmute user';
+			}
+		} else {
+			unminimizeComment(comment);
+			for (const button of select.all('.rgh-mute-unmute-button', comment)) {
+				button.textContent = 'Mute user';
+			}
+		}
+	}
+
+	requestAnimationFrame(() => {
+		const newOffset = comment.getBoundingClientRect().top;
+		window.scrollBy(0, newOffset - viewportOffset);
+	});
+}
+
+async function minimizeMutedUserComments(): Promise<void> {
+	const mutedUsers = await getMutedUsers();
+
+	const comments = select.all('.js-discussion .js-comment-container');
+	for (const comment of comments) {
+		const user = select('.author', comment)!.textContent!;
+		const isMutedUser = mutedUsers.includes(user);
+
+		const dropdowns = select.all('.show-more-popover', comment);
+		for (const dropdown of dropdowns) {
+			// Add option to mute or unmute user
+			appendBefore(dropdown, 'a[data-ga-click^="Report"]',
```

Comment:
Like we do in `faster-comment-minimize` (I don’t remember the actual feature name) we might be able to add the menu item **only when the user clicks on dropdown**, on demand, to avoid changing the DOM of every comment for a feature that’s very rarely used. 

This means would split this function in two parts:

- hide muted comments on load
- add menu item on dropdown open